tests/test_bounce_integral.py

```diff
@@ -0,0 +1,751 @@
+                )
+
+    c = np.array(
+        [
+            [1, 0, 0, 0],
+            [0, 1, 0, 0],
+            [0, 0, 1, 0],
+            [0, 0, 0, 1],
+            [1, -1, -8, 12],
+            [1, -6, 11, -6],
+            [0, -6, 11, -2],
+        ]
+    )
+    root = _poly_root(c.T, sort=True, distinct=True)
+    for j in range(c.shape[0]):
+        unique_roots = np.unique(np.roots(c[j]))
+        root_filter = _filter_not_nan(root[j])
+        assert root_filter.size == unique_roots.size
+        np.testing.assert_allclose(
+            actual=root_filter,
+            desired=unique_roots,
+            err_msg=str(j),
+        )
+    c = np.array([0, 1, -1, -8, 12])
+    root = _filter_not_nan(_poly_root(c, sort=True, distinct=True))
+    unique_root = np.unique(np.roots(c))
+    assert root.size == unique_root.size
+    np.testing.assert_allclose(root, unique_root)
+
+
+@pytest.mark.unit
+def test_poly_der():
+    """Test vectorized computation of polynomial derivative."""
+    quintic = 6
+    c = np.arange(-18, 18).reshape(quintic, 3, -1) * np.pi
+    # make sure broadcasting won't hide error in implementation
+    assert np.unique(c.shape).size == c.ndim
+    derivative = _poly_der(c)
+    for j in range(c.shape[1]):
+        for k in range(c.shape[2]):
+            np.testing.assert_allclose(
+                actual=derivative[:, j, k], desired=np.polyder(c[:, j, k])
+            )
+
+
+@pytest.mark.unit
+def test_poly_val():
+    """Test vectorized computation of polynomial evaluation."""
+
+    def test(x, c):
+        val = _poly_val(x=x, c=c)
+        if val.ndim != max(x.ndim, c.ndim - 1):
+            raise ValueError(f"Incompatible shapes {x.shape} and {c.shape}.")
+        for index in np.ndindex(c.shape[1:]):
+            idx = (..., *index)
+            np.testing.assert_allclose(
+                actual=val[idx],
+                desired=np.poly1d(c[idx])(x[idx]),
+                err_msg=f"Failed with shapes {x.shape} and {c.shape}.",
+            )
+
+    quartic = 5
+    c = np.arange(-60, 60).reshape(quartic, 3, -1) * np.pi
+    # make sure broadcasting won't hide error in implementation
+    assert np.unique(c.shape).size == c.ndim
+    x = np.linspace(0, 20, c.shape[1] * c.shape[2]).reshape(c.shape[1], c.shape[2])
+    test(x, c)
+
+    x = np.stack([x, x * 2], axis=0)
+    x = np.stack([x, x * 2, x * 3, x * 4], axis=0)
+    # make sure broadcasting won't hide error in implementation
+    assert np.unique(x.shape).size == x.ndim
+    assert c.shape[1:] == x.shape[x.ndim - (c.ndim - 1) :]
+    assert np.unique((c.shape[0],) + x.shape[c.ndim - 1 :]).size == x.ndim - 1
+    test(x, c)
+
+
+@pytest.mark.unit
+def test_get_extrema():
+    """Test that these pitch intersect extrema of |B|."""
+    start = -np.pi
+    end = -2 * start
+    k = np.linspace(start, end, 5)
+    B = CubicHermiteSpline(
+        k, np.cos(k) + 2 * np.sin(-2 * k), -np.sin(k) - 4 * np.cos(-2 * k)
+    )
+    B_z_ra = B.derivative()
```

Comment:
Is this an agreed-upon convention? This is B_z at constant rho and alpha, right?

Reply:
Yes, or at least `data["B_z|r,a"]` is agreed convention